web/src/beta/features/PluginPlayground/Plugins/index.tsx

```diff
@@ -34,99 +34,142 @@ const Plugins: FC<Props> = ({
   updateFileTitle,
   deleteFile,
   handleFileUpload,
-  sharedPlugins
+  sharedPlugin
 }) => {
   const [isAddingNewFile, setIsAddingNewFile] = useState(false);
 
   const handleShareIconClicked = (pluginId: string): void => {
     encodeAndSharePlugin(pluginId);
   };
 
+  const customPlugin = presetPlugins.find((plugin) => plugin.id === "custom");
+  const pluginsWithoutCustom = presetPlugins.filter(
+    (plugin) => plugin.id !== "custom"
+  );
+
+  const PluginEntryItem: FC<{
+    plugin: { id: string; title: string };
+    selectedPluginId: string;
+    onSelect: (id: string) => void;
+    onShare: (id: string) => void;
+  }> = ({ plugin, selectedPluginId, onSelect, onShare }) => (
+    <EntryItem
+      key={plugin.id}
+      highlighted={selectedPluginId === plugin.id}
+      onClick={() => onSelect(plugin.id)}
+      title={plugin.title}
+      optionsMenu={[
+        {
+          id: "0",
+          title: "share",
+          icon: "paperPlaneTilt",
+          onClick: () => onShare(plugin.id)
+        }
+      ]}
+      optionsMenuWidth={100}
+    />
+  );
+
   return (
     <Wrapper>
-      <PluginList>
-        {presetPlugins.map((category) => (
-          <div key={category.id}>
-            <CategoryTitle key={category.id}>{category.title}</CategoryTitle>
-            {category.plugins.map((plugin) => (
-              <EntryItem
-                key={plugin.id}
-                highlighted={selectedPlugin.id === plugin.id}
-                onClick={() => selectPlugin(plugin.id)}
-                title={plugin.title}
-                optionsMenu={[
-                  {
-                    id: "0",
-                    title: "share",
-                    icon: "paperPlaneTilt",
-                    onClick: () => handleShareIconClicked(plugin.id)
-                  }
-                ]}
-                optionsMenuWidth={100}
+      <IconList>
+        <Icon icon="addFile" size={32} ariaLabel="Add new file" />
```

Comment:
1. You need to use the `IconButton` component for all `icons` instead of the Icon component because, later, you'll need to handle the onClick event.
2. Can you ask @Red about the Icons? The icons you added are too small, which is why it's difficult to use them in the `IconButton`

Reply:
For the functions we already have (like add file here), could you refactor connecting the function with the new button in this PR as well?

Reply:
Done